cmd/kured/main.go

```diff
@@ -550,17 +547,22 @@ func uncordon(client *kubernetes.Clientset, node *v1.Node) error {
 }
 
 func invokeReboot(nodeID string, rebootCommand []string) {
-	log.Infof("Running command: %s for node: %s", rebootCommand, nodeID)
-
 	if notifyURL != "" {
 		if err := shoutrrr.Send(notifyURL, fmt.Sprintf(messageTemplateReboot, nodeID)); err != nil {
 			log.Warnf("Error notifying: %v", err)
 		}
 	}
 
-	if err := newCommand(rebootCommand[0], rebootCommand[1:]...).Run(); err != nil {
-		log.Fatalf("Error invoking reboot command: %v", err)
+	var booter reboot.Reboot
```

Comment:
I think the cleanest way to evolve the current implementation with this new code is to instantiate our `reboot.Rooter` object in `func root(cmd *cobra.Command, args []string)` and then pass that into `go rebootAsRequired(...)`

That would require the following modifications:

`rebootAsRequired` becomes:

`func rebootAsRequired(nodeID string, booter reboot.Reboot, sentinelCommand []string, window *timewindow.TimeWindow, TTL time.Duration, releaseDelay time.Duration) {`

We invoke `rebootAsRequired` thusly:

`go rebootAsRequired(nodeID, booter, hostSentinelCommand, window, lockTTL, lockReleaseDelay)`

In the flow of `rebootAsRequired`, instead of:

```
                invokeReboot(nodeID, rebootCommand)
		for {
			log.Infof("Waiting for reboot")
			time.Sleep(time.Minute)
		}
```

We do this:

```
                booter.Reboot()
		for {
			log.Infof("Waiting for reboot")
			time.Sleep(time.Minute)
		}
```

And now we can delete the `invokeReboot` func entirely.

What do you think?